code/frameworks/angular/src/server/angular-cli-webpack.js

```diff
@@ -66,6 +66,9 @@ exports.getWebpackConfig = async (baseConfig, { builderOptions, builderContext }
 
       // Options provided by user
       ...builderOptions,
+      styles: builderOptions.styles
+        ?.map((style) => (typeof style === 'string' ? style : style.input))
```

Comment:
@valentinpalkovic hi. I think there is a slight mistake here since first you use `map` to take the string, but the `filter` will always have a string. Now objects with `"inject": false` will not be created as separate files

Reply:
Hi @andriy-statsenko-lightit

Thank you for pointing this out. You're right! Fix incoming: https://github.com/storybookjs/storybook/pull/27108

Reply:
@valentinpalkovic Thanks for the answer. It’s also worth noting that even if you swap the `filter` and `map`, files with `“inject”: false` will not be included in the final build. 🤔

Reply:
Maybe I misunderstood the `inject` property in the past. Are you saying we shouldn't filter at all, but `inject` only means whether the CSS is bundled in or outputted as a separate file?

Reply:
Ok so `inject: false` just excludes the bundle from the injection to `index.html` file, but the file will remain in the folder. `inject: true` includes the file to `index.html`. Example:
<img width="1460" alt="image" src="https://github.com/storybookjs/storybook/assets/69786713/efaae387-d260-46c6-9f87-d4e34d1e9af1">
As I understand it, the problem was that files with `inject: true` were not included in the` index.html `file or were not included in the `main.css `file (not knowing exactly how the storybook builder works). After this fix, they will now be in the `main.css` file, but files with `inject: false` will not be created at all in the build (`dist`) folder.

Reply:
Ok, so do I understand you correctly?:

- If inject: true is set, the CSS file is compiled/transpiled AND should be part of the main bundle
- If inject: false is set, the CSS file is compiled/transpiled but is NOT part of the main bundle. In this case you have to import it somewhere manually? Can you name some examples where this is relevant?

Reply:
Yeah, exactly. So in my case, I use those `inject-false` files in `manager-head/preview-head.html` and import them like CSS or JS files.

Reply:
And did the fix in this PR break this behavior? Or was it never working?

Reply:
Yes, this fix simply filters out files with `inject: false`. As a result, the `main.css` file contains only files that are either files with `inject: true` or with no `inject` (it is `true` by default).
But there is an interesting point - if you comment out this fix, it also does not always work normally. For example, this config creates separate files:
```
"styles": [
               {
                 "input": ".storybook/styles/ui-manager-overrides.scss",
                 "inject": false,
                 "bundleName": "ui-manager-overrides"
               },
               {
                 "input": ".storybook/styles/ui-preview-overrides.scss",
                 "inject": false,
                 "bundleName": "ui-preview-overrides"
               }
             ]
```
<img width="1149" alt="image" src="https://github.com/storybookjs/storybook/assets/69786713/d8ab1416-426e-432a-928f-78dd919a0786">

But if you add `"src/styles/main.scss"`, then only the `main.css` file is created, and files with `"inject": false `are simply not created in the dist folder. This problem only occurs with Storybook Builder; there is no such problem with Angular Builder.
```
"styles": [
              "src/styles/main.scss",
              {
                "input": ".storybook/styles/ui-manager-overrides.scss",
                "inject": false,
                "bundleName": "ui-manager-overrides"
              },
              {
                "input": ".storybook/styles/ui-preview-overrides.scss",
                "inject": false,
                "bundleName": "ui-preview-overrides"
              }
            ]
```
<img width="1216" alt="image" src="https://github.com/storybookjs/storybook/assets/69786713/ea6ce264-16a3-4ccd-85f2-a6a027a99990">

Reply:
That's interesting! Does it work if you define `src/styles/main.scss` in the object notation alongside with the other two scss files, like:

```
"styles": [
              {
                "input": "src/styles/main.scss",
                "bundleName": "main.scss",
                "inject": true
              },
              {
                "input": ".storybook/styles/ui-manager-overrides.scss",
                "inject": false,
                "bundleName": "ui-manager-overrides"
              },
              {
                "input": ".storybook/styles/ui-preview-overrides.scss",
                "inject": false,
                "bundleName": "ui-preview-overrides"
              }
            ]
```

Reply:
Yeah, if there is an object, it works ok, but if there is a path as a string, then files with `“inject”: false` simply do not appear in `dist` folder. This is interesting
<img width="1120" alt="image" src="https://github.com/storybookjs/storybook/assets/69786713/c7496448-d7a5-4b2c-a0a1-7966a328ea4c">

With string path:
<img width="1135" alt="image" src="https://github.com/storybookjs/storybook/assets/69786713/d60858ec-3e20-4d4c-9f13-a001dccf30e0">

Reply:
But that's great. Then I can kind of revert the work of this PR and do an internal mapping of styles defined as strings to the object notation. For the user, nothing changes, but things seem to work then!

Reply:
Yeah, it sounds like a great idea 🙂

Reply:
I've put together a fix here: https://github.com/storybookjs/storybook/pull/27108

@andriy-statsenko-lightit Would you mind to test the following canary release and give me some feedback? `0.0.0-pr-27108-sha-2470133a`

Reply:
@valentinpalkovic yeah, it works fine with different `styles` arrays: string only, string + objects, and objects only. Thanks!